### PR TITLE
feat(tests): Add Cypress test for Agent Configuration 

### PR DESCRIPTION
**Pull request by mdebauchez**
## Description

image

**Tests** #MON-146653 & #MON-147640

### CODE REVIEW

File: centreon/tests/e2e/features/Agent-configuration/03-agent-configuration-deletion/index.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  cy.get('*[role="rowgroup"]')
+    .should('contain', 'telegraf-001');
+});
+
+When('the user deletes the agent configuration', () => {
+  cy.getByTestId({ testId: 'DeleteOutlineIcon' }).click();
+});
+
+When('the user confirms on the pop-up', () => {
+  cy.get('[class^="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary"]').click();
```
Comment (MedWassCh) [inline centreon/tests/e2e/features/Agent-configuration/03-agent-configuration-deletion/index.ts:65]: You should probably find a different selector; it shouldn’t stay like this

File: centreon/tests/e2e/features/Agent-configuration/03-agent-configuration-deletion/index.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  cy.wait('@deleteAgents');
+});
+
+Then('the agent configuration is no longer displayed in the listing page', () => {
+  cy.contains('Welcome to the poller/agent configuration page').should('be.visible');
+  cy.contains('telegraf-001').should('not.exist');
+});
+
+When('the user cancel on the pop-up', () => {
+  cy.get('[class^="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-mcvclk-button"]').click();
```
Comment (MedWassCh) [inline centreon/tests/e2e/features/Agent-configuration/03-agent-configuration-deletion/index.ts:75]: Same comment afn this one contains a dynamic css  css-mcvclk-button

File: centreon/tests/e2e/features/Agent-configuration/05-agent-configuration-check-permissions/index.ts [hunk 3]
```diff
@@ -0,0 +1,313 @@
...
+Then('the updated configuration is displayed correctly in the Agents Configuration page', () => {
+  cy.get('*[role="rowgroup"]')
+    .should('contain', 'telegraf-001-updated');
+  cy.get('*[role="rowgroup"]')
+    .should('contain', '2 pollers');
+});
+
+When('the user deletes the Agents Configuration', () => {
+  cy.getByTestId({ testId: 'DeleteOutlineIcon' }).eq(0).click();
+  cy.get('[class^="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary"]').click();
```
Comment (MedWassCh) [inline centreon/tests/e2e/features/Agent-configuration/05-agent-configuration-check-permissions/index.ts:137]: Same comment

File: centreon/tests/e2e/features/Agent-configuration/05-agent-configuration-check-permissions/index.ts [hunk 4]
```diff
@@ -0,0 +1,313 @@
...
+    .should('contain', 'telegraf-001');
+  cy.get('*[role="rowgroup"]')
+    .should('contain', '2 pollers');
+});
+
+Then('a pop up is displayed with all of the agent configuration information with the 2 pollers', () => {
+  cy.contains('Update poller/agent configuration').should('be.visible');
+  cy.getByLabel({ label: 'Agent type', tag: 'input' }).should('have.value', 'Telegraf');
+  cy.getByLabel({ label: 'Name', tag: 'input' }).should('have.value', 'telegraf-001');
+  cy.get('[class^="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-deletable MuiChip-deletableColorDefault MuiChip-filledDefault MuiAutocomplete-tag MuiAutocomplete-tagSizeSmall"]').eq(0).should('have.text', 'Central');
```
Comment (MedWassCh) [inline centreon/tests/e2e/features/Agent-configuration/05-agent-configuration-check-permissions/index.ts:244]: You need to find something much more precise; it’s best practice not to leave selectors like this

File: centreon/tests/e2e/features/Agent-configuration/05-agent-configuration-check-permissions/index.ts [hunk 5]
```diff
@@ -0,0 +1,313 @@
...
+  cy.get('[class^="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-deletable MuiChip-deletableColorDefault MuiChip-filledDefault MuiAutocomplete-tag MuiAutocomplete-tagSizeSmall"]').eq(1).should('have.text', 'Poller-1');
```
Comment (MedWassCh) [inline centreon/tests/e2e/features/Agent-configuration/05-agent-configuration-check-permissions/index.ts:245]: Same comment

**Review by MedWassCh — APPROVED** [review]

**Review by Thebarda — APPROVED** [review]

**Pull request merged**